packages/lingui-react/src/WithI18n.js

```diff
@@ -41,7 +47,7 @@ export default ({ update = true }: WithI18nOptions = {}) => function<P, C: React
     render () {
       const { i18n } = this.getI18n()
       // $FlowIgnore: https://github.com/facebook/flow/issues/3241
-      return <WrappedComponent {...this.props} i18n={i18n} />
+      return <WrappedComponent ref={(x)=>this.ctrls.wrappedInstance=x} {...this.props} i18n={i18n} />
```

Comment:
Using fat-arrow functions inside `render()` forces re-rendering or `WrappedComponent` because it creates a new function for every render.

`react-redux` (https://github.com/reactjs/react-redux/blob/master/src/components/connectAdvanced.js) binds function in constructor to avoid this, we can use class properties:

```
setWrapperComponent = ref => this.wrappedComponent = ref

render () {
   ...
   return <WrappedComponent ref={this.setWrappedComponent} {...this.props} i18n={i18n} />
}
```